python_modules/libraries/dagster-airlift/dagster_airlift/core/load_defs.py

```diff
@@ -326,3 +331,42 @@ def load_airflow_dag_asset_specs(
         retrieval_filter=retrieval_filter or AirflowFilter(),
     ).get_or_fetch_state()
     return list(spec_iterator(construct_dag_assets_defs(serialized_data)))
+
+
+def uri_to_asset_key(uri: str) -> AssetKey:
+    last_path_segment = uri.split("/")[-1]
+    with_ext_removed = last_path_segment.split(".")[0]
+    return AssetKey(with_ext_removed)
```

Comment:
open to suggestions about what this asset key should actually be

Reply:
I think we should probably include all segments of the URI to avoid name collisions, e.g.:

`s3://foo/bar/baz.py`

would become

`AssetKey(["foo", "bar", "baz"])`

could also be convinced on adding s3 in there.

Reply:
I think s3 should be extracted to a kind

Reply:
recommend `urlparse` here

Reply:
should we allow some customization here?  some translation layer?

Reply:
yea definitely. I think this will be an injectable method on the component